### PR TITLE
Fix Indirect Corrections Monte Carlo crash

### LINKED ISSUE #29277: Indirect Corrections Monte Carlo crash
### Expected behavior
Should be able to save results from Indirect->corrections->monte Carlo
### Actual behavior
Causes Mantid to crash
### Steps to reproduce the behavior
- Go to Indirect -> Corrections -> Calculate Monte Carlo Absorption
- I've been testing with the following parameters
- Sample - iris26176_graphire002_red.nxs
- Events 1000
- Beam height 3.2, width 2.1
- Shape
- Height 1, width 1, thickness 0.1
- Chemical formula H2-O
- Run and then save results, it will crash
### Platforms affected
Presumably all

### PR DESCRIPTION
**Pull request by StephenSmith25**
**Description of work.**
This PR fixes a crash into indirect corrections interface. This crash occurred when the user attempted to save the results from the corrections in the Monte Carlo tab.

**Report to:** ISIS/Spencer

**To test:**

Data to test the interface can be found in - #26631.
- Go to Indirect -> Corrections -> Calculate Monte Carlo Absorption
- I've been testing with the following parameters
- Sample - iris26176_graphire002_red.nxs
- Events 1000
- Beam height 3.2, width 2.1
- Shape
- Height 1, width 1, thickness 0.1
- Chemical formula H2-O
- Run and then save results, should no longer be a crash

Fixes #29277. 

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

File: qt/scientific_interfaces/Indirect/AbsorptionCorrections.cpp [hunk 1]
```diff
@@ -502,14 +502,19 @@ void AbsorptionCorrections::convertSpectrumAxes(
...
+  // The m_saveAlgRunning flag is queried here so the
+  // processWavelengthWorkspace isn't executed at the end of the
+  // saveAlg completing, as this will throw an exception.
+  if (!error && !m_saveAlgRunning) {
     processWavelengthWorkspace();
     setOutputPlotOptionsWorkspaces({m_pythonExportWsName});
+  } else if (!error && m_saveAlgRunning) {
+    setButtonsEnabled(true);
+    m_saveAlgRunning = false;
   } else {
```
Comment (clayton-tom) [inline qt/scientific_interfaces/Indirect/AbsorptionCorrections.cpp:518]: Should `m_saveAlgRunning` be set to false in this case also? Seems like a `true` state could propagate to the next run if an error occurred

Comment (clayton-tom) [inline qt/scientific_interfaces/Indirect/AbsorptionCorrections.cpp:518]: Should `m_saveAlgRunning` be set to `false` in this case also? Seems like a `true` value could propagate to the next run if an error occurred in the algorithm

Comment (clayton-tom) [reply qt/scientific_interfaces/Indirect/AbsorptionCorrections.cpp:518]: For clarity as I can't seem to select the right lines - I'm on about the final `else` statement, sorry for the confusion

Comment (StephenSmith25) [reply qt/scientific_interfaces/Indirect/AbsorptionCorrections.cpp:518]: Ah yeah thats a good point.

**Review by clayton-tom — APPROVED** [review]
Code looks good and ws saves fine after the run 👍

**Pull request merged**